core/dbt/parser/util.py

```diff
@@ -254,19 +255,26 @@ def process_sources(cls, manifest, current_project: str):
         return manifest
 
     @classmethod
-    def add_new_refs(cls, manifest, current_project: Project, node, macros):
-        """Given a new node that is not in the manifest, copy the manifest and
-        insert the new node into it as if it were part of regular ref
-        processing
+    def add_new_refs(cls, manifest, config: RuntimeConfig, node, macros):
+        """Given a new node that is not in the manifest, insert the new node
+        into it as if it were part of regular ref processing.
         """
-        manifest = manifest.deepcopy()
+        if config.args.single_threaded or SINGLE_THREADED_HANDLER:
```

Comment:
do you want to use `_single_threaded` defined below here?

Reply:
ugh, oops, no - I want to remove that function.